client/src/main/python/slipstream/executors/node/NodeDeploymentExecutor.py

```diff
@@ -112,10 +112,11 @@ def _execute_build_recipes(self):
         packages = self.node_instance.get_packages()
         if packages:
             message = 'Installing packages: %s' % ', '.join(packages)
+            fail_msg = "Failed installing packages on '%s'" % self._get_node_instance_name()
             util.printStep(message)
             self.wrapper.set_statecustom(message)
             cmd = util.get_packages_install_command(self.node_instance.get_platform(), packages)
-            self._run_target_script('#!/bin/sh -xe\n%s' % cmd)
+            self._launch_script('#!/bin/sh -xe\n%s' % cmd, fail_msg=fail_msg)
         else:
             util.printStep('No packages to install')
 
```

Comment:
I would suggest to extract lines 112-121 to a new method: _install_user_packages()